src/client.rs

```diff
@@ -0,0 +1,81 @@
+//! Clients for Google Cloud Storage endpoints.
+
+use tokio::sync::Mutex;
+
+use crate::token::Token;
+
+mod bucket;
+mod bucket_access_control;
+mod default_object_access_control;
+mod hmac_key;
+mod object;
+mod object_access_control;
+
+pub use bucket::BucketClient;
+pub use bucket_access_control::BucketAccessControlClient;
+pub use default_object_access_control::DefaultObjectAccessControlClient;
+pub use hmac_key::HmacKeyClient;
+pub use object::ObjectClient;
+pub use object_access_control::ObjectAccessControlClient;
+
+/// The primary entrypoint to perform operations with Google Cloud Storage.
+pub struct Client {
+    client: reqwest::Client,
+
+    /// Static `Token` struct that caches
+    token_cache: Mutex<Token>,
+}
+
+impl Default for Client {
+    fn default() -> Self {
+        Self {
+            client: Default::default(),
+            token_cache: Mutex::new(Token::new(
+                "https://www.googleapis.com/auth/devstorage.full_control",
+            )),
+        }
+    }
+}
+
+impl Client {
```

Comment:
It might be sensible to include a function in `Client` called `new`, that just delegates to `Client::default`, because people will expect to be able to call `new` and get a new `Client`

Reply:
Yep, I debated that. I tend to just to `default` in my _own_ code because I'm lazy ;-). Fixed.

Reply:
Brilliant